src/GT.Core/Services/Impl/GTListingService.cs

```diff
@@ -213,53 +214,46 @@ public async Task<List<ListingOverviewDTO>> GetAsync(IListingFilterModel? filter
 				.Where(k => k != null)
 				.ToArray();
 
-			if (keywords is not null && keywords.Any())
+			if (keywords is not null 
+				&& query is not null 
+				&& keywords.Any())
 			{
-				foreach (var keyword in keywords)
+				for(int i = 0; i < keywords.Length; i++)
 				{
+					var keyword = keywords[i].Replace('_', ' ');
```

Comment:
Why is this step necessary?

Reply:
It's there to allow users to search for full phrases as one keyword, with underscores instead of spaces since the method splits keywords by spaces. For example, a user could search for "software_engineer" in order to only match job postings that contain the exact phrase "software engineer", rather than having the filter method look for "software" and "engineer" individually.

Reply:
Maybe worth considering to swap '_' for '+' instead - applying same convention as in google searches that people are more used to?

Reply:
I am fine with both the current and suggested implementation.

Reply:
Either is fine with me, maybe it is something we can discuss in a future meeting?

Reply:
Agreed. Leave as is for now.